doc/HLD_Version_Numbers.rst

```diff
@@ -0,0 +1,328 @@
+======================================
+High Level Design of Version Numbers
+======================================
+
+This document presents a high level design (HLD) of version numbers in Motr M0 core. The main purposes of this document are: (i) to be inspected by M0 architects and peer designers to ascertain that high level design is aligned with M0 architecture and other designs, and contains no defects, (ii) to be a source of material for Active Reviews of Intermediate Design (ARID) and detailed level design (DLD) of the same component, (iii) to serve as a design reference document.
+
+***************
+Introduction
+*************** 
+
+Version numbers identify particular file system states and order updates thereof. They are used for distributed transaction management, concurrency control (both lock based and optimistic) and object history navigation.
+
+A version number is stored together with the file system state whose version it identifies. Specifically, multiple cached replicas of the same state are all tagged with version numbers matching each other in the sense defined below. Version numbers are treated as a distributed resource. A piece of file system state to which a version number is assigned is called a unit. Unit granularity can be a file system object or a part of a file system object. 
+
+***************
+Definitions
+*************** 
+
+See the Glossary for general M0 definitions and HLD of FOL for the definitions of file system operation, update and lsn. The following additional definitions are required:
+
+- for the purposes of the present design it is assumed that a file system update acts on units [1]. For example, a typical meta-data update acts on one or more "inodes" and a typical data update acts on inodes and data blocks. Inodes, data blocks, directory entries, etc. are all examples of units. It is further assumed that units involved in an update are unambiguously identified [2] and that complete file system state is a disjoint union of states of comprising units. (Of course, there are consistency relationships between units, e.g., inode nlink counter must be consistent with contents of directories in the name-space).
```

Comment:
>- for the purposes of the present design it is assumed that a file system update acts on units [1].

I am sorry, in the last review comments, I said this should be changed to [1].
But this was wrong.
When I looked at the original doc, this should be a link to some internal requirement.
Please check with the original doc to correct all the internal links.

And keep the [0] and [1] of internal links listed at the end of this document.

Reply:
Please follow and compare with the original document to correct these.

It is not a copy & paste to convert a doc.

Reply:
Incorporated.